app/models/APIResponse.scala

```diff
@@ -0,0 +1,31 @@
+package models
+
+import com.amazonaws.services.dynamodbv2.model.AmazonDynamoDBException
+import com.gu.atom.data.DataStoreError
+import play.api.Logger
+import io.circe._
+import io.circe.syntax._
+import play.api.mvc._
+import io.circe.generic.semiauto._
+
+case class AtomWorkshopAPIResponse(message: String)
+object AtomWorkshopAPIResponse{
+  implicit val atomWorkshopApiResponseEncoder: Encoder[AtomWorkshopAPIResponse] = deriveEncoder[AtomWorkshopAPIResponse]
+}
+
+
+object APIResponse extends Results {
+  def exceptionToResult(e: Exception) = {
+    Logger.error(e.getMessage, e)
+    InternalServerError(AtomWorkshopAPIResponse(e.getMessage).asJson.noSpaces)
+  }
+
+  def apply[T](result: Either[Exception, T])(implicit encoder: Encoder[T]): Result = {
+    val res = result.fold({
+      case e: DataStoreError => exceptionToResult(e)
```

Comment:
Since you handle all these the same why, why do this match? Surely the following would suffice?

    exceptionToResult(_)